src/Design.cc

```diff
@@ -208,6 +208,35 @@ bool Design::isSequential(odb::dbMaster* master)
   return lib_cell->hasSequentials();
 }
 
+bool Design::isInClock(odb::dbInst* inst) {
+  for (auto* iterm : inst->getITerms()) {
+    auto* net = iterm->getNet();
+    if (net != NULL && net->getSigType() == odb::dbSigType::CLOCK){
+      return true;
+    }
+  }
+  return false;
+}
+
+std::uint64_t Design::getNetRoute(odb::dbNet* net) {
```

Comment:
getNetRoute is rather vague, perhaps getNetRoutedLength?

Btw, what would you do with the power supply wire length?